cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -180,6 +182,8 @@ protected void configure() {
                         e.getFailures().iterator().next().getFullMessage()), e);
     }
 
+    applicationConfigurer.emitAppMetadata(new Metadata(Collections.emptyMap(), spec.getConnectionsUsed()));
```

Comment:
Here we emit the connection ids with tags. Because of this, the connection is by default assumed to be in the same namespace of the pipeline(cross namespace will make relation bad). It also adds some burden on supporting system scope connection post 6.5. 
What we can do is:
1. For system scope connection, we restrict the length of id to 48 and add '-S' to indicate it is system scope
2.Use property to save the connection, but the key has a reserved key "tags", we can do some special handling on generating the id if a connection name is "tags", but it will look strange since we have standards for a specific name. 
@albertshau what do you think is a better approach?

Reply:
I think option 1 is better, though to be consistent I think we would want to make the length limit the same for all connections instead of allowing a slightly longer one for namespaced connections.

Reply:
It would also be worth looking into this length limit at some point. A max length of 50 seems pretty arbitrary.